### PR TITLE
Changes to make this DLL pass immutability analysis

### PR DESCRIPTION
**Pull request by j3parker**

### CODE REVIEW

**Comment by j3parker** [comment]
revolutionary!

**Comment by j3parker** [comment]
Ah I sliced up these changes dumb; better to wait for .NET 4.6.1 before this one.

**Comment by j3parker** [comment]
@omsmith aight, second look? totally will build this time

**Comment by j3parker** [comment]
GAHHHH

**Comment by j3parker** [comment]
Fixed - needed to publish the package.

File: src/D2L.Security.OAuth2/packages.config [hunk 1]
```diff
@@ -3,5 +3,7 @@
   <package id="D2L.Services.Core.Exceptions" version="1.0.0.27289" targetFramework="net451" />
   <package id="D2L.Services.Core.Extensions" version="1.1.1.0" targetFramework="net451" />
   <package id="Newtonsoft.Json" version="6.0.8" targetFramework="net451" />
+  <package id="System.Collections.Immutable" version="1.4.0" targetFramework="net461" />
```
Comment (JeffAshton) [inline src/D2L.Security.OAuth2/packages.config]: Mismatch with the LP, 

http://search.dev.d2l/source/search?&q=%22System.Collections.Immutable%22&path=packages.config&project=Lms&sort=fullpath

Comment (JeffAshton) [reply src/D2L.Security.OAuth2/packages.config]: Not sure if that's going to cause you troubles

Comment (j3parker) [reply src/D2L.Security.OAuth2/packages.config]: Interesting. I don't 100% remember what happens in the end either.

Any objections to updating LP?

Comment (JeffAshton) [reply src/D2L.Security.OAuth2/packages.config]: None from me.  That's at least the safer direction right?

Comment (JeffAshton) [reply src/D2L.Security.OAuth2/packages.config]: You want the later version of the DLL deployed, since it's a minor, other code should still be good.

**Pull request merged**